src/app/public/src/modules/table-of-contents/table-of-contents.component.spec.ts

```diff
@@ -67,40 +92,214 @@ describe('StacheTableOfContentsComponent', () => {
     expect(fixture).toExist();
   });
 
-  it('should detect changes after init', () => {
-    const spy = spyOn((component as any).cdr, 'markForCheck');
-    component.ngAfterViewInit();
-    expect(spy).toHaveBeenCalled();
+  it('should update routes on scroll, undefined routes', () => {
```

Comment:
The statements here should reflect what the test should be doing/expecting so it's obvious when reading test cases.  e.g `should not call updateView when routes are undefined.`